### ISSUE TITLE
SyntaxError: unknown: 'this' is not allowed before super()

### ISSUE THREAD

**Issue by bilkusg** [fsprojects/Fable#124]
### Description

The following f# code does not compile with fable

    type TestClass(s:string) =
        let mys = s
        member x.showS() = Console.WriteLine (s) 

    type TestDerivedClass(i:int) as this  =
        inherit TestClass(string i)
        do this.showS()

### Repro steps

Create a fable .fs file with the above definition and attempt to compile it using fable

### Expected behavior

Should compile into valid javascript

### Actual behavior

Get error message:

    { [SyntaxError: unknown: 'this' is not allowed before super() (This is an error on an internal node.     Probably an internal error)] _babel: true }

### Known workarounds

None

* Operating system: Windows 10
* Branch: Current npm package as of 2016 May 14th
* .NET Runtime, CoreCLR or Mono Version 4.6

**Comment by alfonsogarciacaro** [comment]
This is problematic. I've faced this issue in the past and my naive assumption was that nobody would want to use `this  in the constructor :) The problem here is the compiler adds several checkers in this situation which are not compatible with JS and also the generated code is variable and difficult to identify in pattern matching.

I'll try to come up with something. For the moment there's a (dirty) workaround which is using `Emit` to cheat the F# compiler when using `this` as it's used in the [React helper](https://github.com/fsprojects/Fable/blob/master/import/react/Fable.Helpers.React.fs#L574-L579):

```
type TestClass(s:string) =
    let mys = s
    member x.showS() = Console.WriteLine (s) 

[<Emit("this")>]
let private this = obj()

type TestDerivedClass(i:int)  =
    inherit TestClass(string i)
    do this?showS $ ()
```

**Comment by bilkusg** [comment]
Ah OK. I had seen that code in the helper and wondered what it was about.

If you are a functional programming purist, ( which I try to be **mainly** ), then obviously there is no
need for imperative constructor blocks like this. 
But in the real world, you may need to interface to something  out there through a class with mutable members, and then these do blocks are kind of important. Indeed, the fact that you needed the workaround in React.Helper shows what I mean.

I appreciate the difficulty, and while I think it would be great if you could find a clean solution, for the moment a brief update to the Interacting.md document might be helpful. Even just a reference to the error message and this issue..

**Comment by alfonsogarciacaro** [comment]
I found a solution though a bit hacky. Some _superfluous_ code is still generated but the compilation should work. The fix has been published in version 0.2.12. Github closed the issue automatically but if it still doesn't work for you please feel free to open it. Cheers!
